src/base/base.c

```diff
@@ -608,23 +611,30 @@ static void collect_params(function_stack_entry *fse, zend_execute_data *zdata,
 #endif
 }
 
-function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_array *op_array, int type)
+function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_array *op_array, int type, int use_current_execute_data)
```

Comment:
This function is reused to reconstruct the stack when jit debugging is started. We will call it successively with the execute data for each stack frame. To be able to do this, we add a new parameter that tells it if it needs to use the current_execute_data or the passed zdata to record the data for that frame